### PR TITLE
[2.7] Remove BC api version in Validator

### PR DESCRIPTION
**Pull request by saro0h**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |

Linked to PR https://github.com/symfony/symfony/pull/13338#issuecomment-69344601.

### CODE REVIEW

File: src/Symfony/Component/Validator/Validation.php [hunk 1]
```diff
@@ -30,12 +29,6 @@
     const API_VERSION_2_5 = 2;
 
     /**
-     * The Validator API provided by Symfony 2.5 and newer with a backwards
-     * compatibility layer for 2.4 and older.
-     */
-    const API_VERSION_2_5_BC = 3;
```
Comment (Tobion) [inline src/Symfony/Component/Validator/Validation.php]: BC break

**Comment by saro0h** [comment]
Ping @fabpot : Can you take a look at this one ?

**Comment by fabpot** [comment]
@saro0h Unfortunately, I fail to explain what needed to be done. I won't have time today but let's talk about this one on Monday.

**Comment by saro0h** [comment]
Sure

File: src/Symfony/Component/Validator/ValidatorBuilder.php [hunk 2]
```diff
@@ -321,32 +321,11 @@ public function setPropertyAccessor(PropertyAccessorInterface $propertyAccessor)
     /**
      * {@inheritdoc}
      */
-    public function setApiVersion($apiVersion)
```
Comment (stof) [inline src/Symfony/Component/Validator/ValidatorBuilder.php]: This cannot be removed (BC break)

File: src/Symfony/Component/Validator/Validation.php [hunk 3]
```diff
@@ -20,22 +20,10 @@
...
     /**
-     * The Validator API provided by Symfony 2.5 and newer.
-     */
-    const API_VERSION_2_5 = 2;
-
-    /**
-     * The Validator API provided by Symfony 2.5 and newer with a backwards
-     * compatibility layer for 2.4 and older.
-     */
-    const API_VERSION_2_5_BC = 3;
```
Comment (stof) [inline src/Symfony/Component/Validator/Validation.php]: these cannot be removed (BC break)

**Comment by stof** [comment]
@saro0h what needs to be done is moving everything which is in the LegacyValidator to its parent class RecursiveValidator directly (having the BC layer in the new class directly thanks to the bump to PHP 5.3.9+). The LegacyValidator would then become a deprecated child of the RecursiveValidator with no extra code in it.

The 2.5 and 2.5-bc APIs would then be exactly the same (as we would be able to put the BC layer in 2.5 directly). The concept of API level needs to be kept for the case of the 2.4 API (using it is already deprecated)

**Comment by saro0h** [comment]
Here you go, I did it here: https://github.com/symfony/symfony/pull/13458

I close this PR as, hopefully, I will be able to do this work in 3.0.

**Pull request closed without merging**